grails-app/assets/javascripts/streama/templates/dash.tpl.htm

```diff
@@ -1,13 +1,20 @@
 
 <div class="dashboard">
+  <!--------------------------new releases--------------------------->
   <div ng-if="vm.newReleases.length && !vm.selectedGenre && !vm.isDashSectionHidden('new-releases')">
     <h3>{{'DASHBOARD.NEW_RELEASES' | translate}}</h3>
 
-
     <div uis-owl-carousel class="new-releases-slider" >
       <div ng-repeat="newRelease in vm.newReleases" class="new-releases-slide">
         <img class="poster-image" ng-src="https://image.tmdb.org/t/p/w342/{{newRelease.media.poster_path}}" class="pull-left">
         <div class="image-background" style="background-image: url('https://image.tmdb.org/t/p/w1280/{{newRelease.media.backdrop_path}}')">
+          <div ng-if="!newRelease.media.inWatchlist">
+            <i class="add-icon ion-android-add-circle" ng-click="vm.addToWatchlist(newRelease.media)"></i>
+          </div>
+          <div ng-if="newRelease.media.inWatchlist">
+            <i class="remove-icon ion-android-remove-circle" ng-click="vm.removeFromWatchlist(newRelease.media)"></i>
+          </div>
```

Comment:
All occurences of this combination with two buttons shown with ng-ifs can be refactored into a single visible button with ng-class on it, and using one method (something like vm.updateWatchlist()) to handle either the deletion or adding of a video from/to a watchlist. Basically removes half the watchers and easier to adapt in the future if changes need to be made